### ISSUE TITLE
Support aliases

### ISSUE THREAD

**Issue by patrickheeney** [pnpm/pnpm#117]
`jspm` supports aliases for client side dependencies, would be a great addition to `pnpm`. 

This is how it looks with `jspm`:

```
{
  "my-alias": "npm:pnpm@1.0.0",
  "another-alias": "github:twbs/bootstrap#master"
}

// ./node_modules/my-alias -> ./store/pnpm@1.0.0
```

The store could also be modified to better support other registries like github:

```
./node_modules/.store
  |_ github
  |_ npm
```

To main backwards compat, it would always default to npm and you could introduce a new `package.json` entry to support the new syntax:

```
{
  "pnpm": {
    "dependencies": {
      "pnpm": "npm:pnpm@1.0.0"
    },
    "devDependencies": {}
 }
}  
```

**Comment by zkochan** [comment]
@patrickheeney pnpm will soon support aliases. Here's the PR: https://github.com/pnpm/supi/pull/16

**Comment by zkochan** [comment]
Published in [v1.23.0](https://github.com/pnpm/pnpm/releases/tag/v1.23.0)
